src/qt/walletmodel.cpp

```diff
@@ -861,19 +861,11 @@ void WalletModel::listCoins(std::map<QString, std::vector<COutput> >& mapCoins)
     std::vector<COutput> vCoins;
     wallet->AvailableCoins(&vCoins, nullptr, filter);
 
-    LOCK(wallet->cs_wallet);
     for (const COutput& out : vCoins) {
-        COutput cout = out;
-
-        while (wallet->IsChange(cout.tx->vout[cout.i]) && cout.tx->vin.size() > 0 && wallet->IsMine(cout.tx->vin[0])) {
-            if (!wallet->mapWallet.count(cout.tx->vin[0].prevout.hash)) break;
-            cout = COutput(&wallet->mapWallet[cout.tx->vin[0].prevout.hash], cout.tx->vin[0].prevout.n, 0, true, true);
-        }
-
         CTxDestination address;
-        if (!out.fSpendable || !ExtractDestination(cout.tx->vout[cout.i].scriptPubKey, address))
+        if (!out.fSpendable || !ExtractDestination(out.tx->vout[out.i].scriptPubKey, address))
```

Comment:
As already discussed, this removal changes the behavior (change addresses are no longer stored inside mapCoins using the addressbook address that "originated" the change, but using the actual change address).
I think this is fine, since we removed the account system.